src/routes/Profile/Profile.tsx

```diff
@@ -125,12 +125,13 @@ const Profile: React.FunctionComponent = () => {
         ...prev,
         [variant]: 'claimed',
       }));
-      setCollectionBadges((prev) => {
-        return prev.map((b) => ({
-          ...b,
-          claimedAt: b.variant === variant ? Date.now().valueOf() : b.claimedAt,
-        }));
-      });
+      const nextCollectionBadges = collectionBadges.map((b) => ({
+        ...b,
+        claimedAt: b.variant === variant ? Date.now().valueOf() : b.claimedAt,
```

Comment:
what is this `nextCollectionBadges` used for? is this parsing through all awarded badges and setting `claimedAt` either at `now` or `time of claiming`?

Reply:
`nextCollectionBadges` is just temporary variable in this context, used to determine what is the next state for the collection badges. Since I don't want to "requery" the subgraph once again I edit the already in-state array.